src/isomorphic/classic/element/ReactElementValidator.js

```diff
@@ -181,6 +181,31 @@ function validatePropTypes(element) {
   }
 }
 
+/**
+ * Given an element, freeze `type` and `defaultProps` (Only used in DEV)
+ *
+ * @param {ReactElement} element
+ */
+function freezeTypeAndDefaultProps(element) {
```

Comment:
@donavon since this is my first PR, i ran the compare_size task as a sanity check...

```bash
Running "compare_size:files" (compare_size) task

   raw     gz Compared to master @ 34c4474472d6877f970c3763de4769d2b98718ab
     =      = build/react-dom-server.js
     =      = build/react-dom-server.min.js
  +725   +177 build/react-dom.js
     =      = build/react-dom.min.js
  +725   +139 build/react-with-addons.js
     =      = build/react-with-addons.min.js
  +725   +137 build/react.js
     =      = build/react.min.js
```

this confirms that build size is only being affected outside `production` 👍

`__DEV__` becomes `process.env.NODE_ENV